test/support/functional_test.rb

```diff
@@ -1,6 +1,17 @@
 # frozen_string_literal: true
 
 class FunctionalTest < ActiveSupport::TestCase
+  setup do
+    Thread.current[:chat_completion_response] = nil
+    Thread.current[:current_step_name] = nil
+    Thread.current[:workflow_metadata] = nil
+    Thread.current[:workflow_context] = nil
+    Thread.current[:step] = nil
+    Thread.current[:result] = nil
+    Thread.current[:error] = nil
+    Roast::Helpers::Logger.reset
```

Comment:
I have no idea what the logger is doing, but it will stop outputting the results of tool calls after running one test. wtf.